Document/0x04f-Testing-Network-Communication.md

```diff
@@ -399,7 +399,16 @@ Some Android and iOS versions do not support some of the recommended cipher suit
 
 ### Static Analysis
 
-Identify all API/web service requests in the source code and ensure that no plain HTTP URLs are used. Make sure that sensitive information is sent over secure channels by using [HttpsURLConnection](https://developer.android.com/reference/javax/net/ssl/HttpsURLConnection.html "HttpsURLConnection") or [SSLSocket](https://developer.android.com/reference/javax/net/ssl/SSLSocket.html "SSLSocket") (for socket-level communication using TLS).
+Since Android 9 (API level 28), cleartext HTTP traffic is blocked by default. However, there are multiple ways in which an application can still send cleartext HTTP traffic:
+
+- Setting the [`android:usesCleartextTraffic`](https://developer.android.com/guide/topics/manifest/application-element#usesCleartextTraffic "Android documentation - usesCleartextTraffic flag") attribute of the `<application>` tag in the AndroidManifest.xml file. Note that this flag is ignored in case a network security config is configured.
+- Configuring the network security config to enable cleartext traffic by setting the cleartextTrafficPermitted attribute to true on `<domain-config>` elements.
+- Using low level libraries (e.g. [`Socket`](https://developer.android.com/reference/java/net/Socket "Socket class") to set up a custom HTTP connection.
+- Using an cross-platform framework, as these typically have their own implementations for HTTP libraries.
+
+Even if one of the above cases is true, it doesn't mean that the application actually sends HTTP traffic. It's possible that an exception is configured, but only HTTPS traffic is used in practice.
```

Comment:
I'm not sure if this line could be a bit confusing. You say this because e.g. only HTTPS URLs can be found in the code (?). But there could be multiple cases where something wrong can happen, e.g. what if the app loads code dynamically and that code contains HTTP URLs?

I think the points in the list make clear that you must not have those flags enabled. But we should still warn that even if they are not enables things can still go wrong. Suggestion here (I'm assuming that the current next line 411 would be moved to be the first line of the current section, so this links to line 413):

```suggestion
All of the above cases must be carefully analyzed as a whole. For example, even if the app does not permit cleartext traffic in its Android Manifest or Network Security Configuration, it might actually still be sending HTTP traffic. That could be the case if it's using a low-level API (for which Network Security Configuration is ignored) or a badly configured cross-platform framework.
```